### ISSUE TITLE
Profiler changes structural type of columns

### ISSUE THREAD

**Issue by kyao** [usc-isi-i2/dsbox-cleaning#53]
The profiler modifies the structural type of columns from string to int, if the semantic type is Integer. It should not change the content of the dataframe.

If recall, we had to do this for the evaluation because the cast_to_type primitive was not working properly. Now, that the evaluation is over. We should undo this and either use the cast_to_type primitive or write our own.

**Comment by RqS** [comment]
@kyao By `It should not change the content of the dataframe.` Is that means we should still `modifies the structural type of columns`, but only metadata, not `the content of the dataframe`

**Comment by kyao** [comment]
It should not structural type of the dataframe column, i.e. leave it as type `str`. And, it should not change the metadata `structural_type`, i.e. leave it as `str`.

**Comment by RqS** [comment]
@kyao Found the followings:
1. Our mean imputer is using semantic type for calculating mean value. So if the actual dtype of a column is not same as its semantic type, it may cause error.
2. Our `cleaning_featurizer` has this function: https://github.com/usc-isi-i2/dsbox-cleaning/blob/02b086870b07977686f58079e30466dc1997282c/dsbox/datapreprocessing/cleaner/cleaning_featurizer.py#L330. I think I added this but forgot for what reason, maybe for letting imputer happy. This function does something like cast_to_type, it detect categorical first, if not, cast the column to int or float, change `semantic_type` and `structural type` consistently. So after `cleaning_featurizer`, it seems column's `semantic_type` and `structural_type` will be consistent and also our imputer will work properly.
3. Disabled the codes in `profiler` which will update `structural_type`, system seems like work as it was before. I made the change: https://github.com/usc-isi-i2/dsbox-cleaning/commit/2a931b402c2243cb2e2331d72752fb4da7fc71df
4. `cast_to_type` primitive change everything to `float`.

We may need the following:
1. Using `cast_to_type` primitive to change everything to `float` maybe OK since it is still numeric. Do not know if it is OK if we represent categorical value as `float` and pass it to model step (classifier or regressor).
2.  `cast_to_type` allows for setting `exclude_columns` hyperparams, we can put integer columns here so it will not perform on.
3. We can come up with our own `cast_to_type` primitive will do more smart things to column based on their different types, integer, float, text. Then we use this instead of common `cast_to_type`.
